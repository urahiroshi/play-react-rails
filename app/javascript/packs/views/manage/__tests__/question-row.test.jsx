import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import QuestionRow from '../question-row.jsx'

const createComponent = () => (
  <QuestionRow
    order={1}
    question={{ id: 1, content: 'hoge', answer: 'fuga' }}
    commands={{}}
    updateQuestion={() => {}}
    deleteQuestion={() => {}}
  />
)

describe('#render', () => {
  it('should show initial view', () => {
    const component = renderer.create(createComponent())
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('when "Edit" button clicked', () => {
  it('should show editing view', () => {
    const wrapper = shallow(createComponent())
    wrapper.find('QuestionRowButtons').props().onClickEdit()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('when "Cancel" button clicked', () => {
  it('should show initial view', () => {
    const wrapper = shallow(createComponent())
    wrapper.find('QuestionRowButtons').props().onClickEdit()
    wrapper.find('EditingButtons').props().onClickCancel()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
