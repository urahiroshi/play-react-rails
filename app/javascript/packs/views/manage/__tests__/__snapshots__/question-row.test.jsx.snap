// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should show initial view 1`] = `
<tr>
  <td>
    1
  </td>
  <td
    className="question-content-cell"
  >
    <div
      className="manage-question-content-preview"
      dangerouslySetInnerHTML={
        Object {
          "__html": "hoge",
        }
      }
    />
  </td>
  <td
    className="question-answer-cell"
  >
    <div
      className="manage-question-answer-preview"
    >
      fuga
    </div>
  </td>
  <td>
    <div>
      <button
        className="btn btn-warning"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="btn btn-danger"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </td>
</tr>
`;

exports[`when "Cancel" button clicked should show initial view 1`] = `
<tr>
  <td>
    1
  </td>
  <td
    className="question-content-cell"
  >
    <div
      className="manage-question-content-preview"
      dangerouslySetInnerHTML={
        Object {
          "__html": "hoge",
        }
      }
    />
  </td>
  <td
    className="question-answer-cell"
  >
    <div
      className="manage-question-answer-preview"
    >
      fuga
    </div>
  </td>
  <td>
    <QuestionRowButtons
      onClickDelete={[Function]}
      onClickEdit={[Function]}
    />
  </td>
</tr>
`;

exports[`when "Edit" button clicked should show editing view 1`] = `
<tr>
  <td>
    1
  </td>
  <td
    className="question-content-cell"
  >
    <div
      className="manage-question-content-preview"
      dangerouslySetInnerHTML={
        Object {
          "__html": "hoge",
        }
      }
    />
  </td>
  <td
    className="question-answer-cell"
  >
    <div
      className="manage-question-answer-preview"
    >
      fuga
    </div>
  </td>
  <td>
    <QuestionRowButtons
      onClickDelete={[Function]}
      onClickEdit={[Function]}
    />
  </td>
</tr>
`;
